libs/user-identity-service/src/test/java/uk/gov/di/ipv/core/library/service/UserIdentityServiceTest.java

```diff
@@ -1352,4 +1586,74 @@ private void mockParamStoreCalls(Map<ConfigurationVariable, String> params) {
         params.forEach(
                 (key, value) -> when(mockConfigService.getSsmParameter(key)).thenReturn(value));
     }
+
+    private String createCredentialWithNameAndBirthDate(
+            String givenName, String familyName, String birthDate) throws Exception {
+        var birthDateList = new ArrayList<String>();
+        birthDateList.add(birthDate);
+        return createCredentialWithNameAndBirthDate(
+                givenName, null, familyName, birthDateList, true);
+    }
+
+    private String createCredentialWithNameAndBirthDate(
+            String givenName, String middleName, String familyName, String birthDate)
+            throws Exception {
+        var birthDateList = new ArrayList<String>();
+        birthDateList.add(birthDate);
+        return createCredentialWithNameAndBirthDate(
+                givenName, middleName, familyName, birthDateList, true);
+    }
+
+    private String createCredentialWithNameAndBirthDate(
+            String givenName, String familyName, String birthDate, boolean isSuccessful)
+            throws Exception {
+        var birthDateList = new ArrayList<String>();
+        birthDateList.add(birthDate);
+        return createCredentialWithNameAndBirthDate(
+                givenName, null, familyName, birthDateList, isSuccessful);
+    }
+
+    private String createCredentialWithNameAndBirthDate(
+            String givenName, String familyName, List<String> birthDates) throws Exception {
+        return createCredentialWithNameAndBirthDate(givenName, null, familyName, birthDates, true);
+    }
+
+    private String createCredentialWithNameAndBirthDate(
```

Comment:
Yep, I'll link something on that ticket.

It would be nice to have a lib for the generation of testing VCs, which can be used across all the lambdas tests.

I think there is also value in having the hardcoded test fixtures, but they need to be up to date and easily viewable...

Reply:
Thanks - I think having a preset list of 'useful test VCs' is good, but we could still generate them with the shared lib rather than using a hardcoded string?

```
public static String DCMAW_BIOMETRIC_PASSPORT_VC = createTestVc(...);
```
etc.

Reply:
Yeah that's possible. I think the thing in my head is that the hardcoded stuff should be examples given to us (or gathered) of *actual* VCs issued by the CRIs. So that we can be sure that core will work with them, or catch issues in them.
The fixtures we currently have aren't really that - lots of incomplete of just wrong examples in there which makes me worry a little that we could easily miss something when using them.
That's my concern about generating them with the lib - it'd be our interpretation of what they should be, and will be more fallible.

Some of this stuff will just naturally get tested by the acceptance tests, but then we don't run as many scenarios (rightly) in staging. I'm not sure what the right balance is...

Reply:
Hmm, I think the difficulty will be keeping those updated.

I suppose in principle this is where the contract tests could help, but we're struggling to really leverage them against the VC structure, which is the important part.

If we generate classes from the JSON schema then I'd feel a lot more confident we're handling the right _structure_, and we're just left with issues around specific values. This shouldn't be such a big deal - we have a lot of specs for this stuff, and if we tear out all the existing evidence parsing (where it deduces the type of VC from the scores present..!) it should become much more robust against different values.